packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -147,6 +151,22 @@ const ProductTile = (props) => {
         })
     }, [product, selectableAttributeId, selectableAttributeValue])
 
+    // Retrive product badges
+    let filteredLabels = new Map()
+    if (product?.representedProduct) {
+        // validate the provided badge labels against the product custom properties. This will allow users to use any boolean custom properties as badges
+        filteredLabels = badgeLabels.reduce((map, item) => {
+            if (
+                item.propertyName &&
+                typeof product.representedProduct[item.propertyName] === 'boolean' &&
+                product.representedProduct[item.propertyName] === true
+            ) {
+                map.set(intl.formatMessage(item.label), item.color)
```

Comment:
What happens if the label of the non-english language that is not defined here since we are using a dynamic key here? You can test that by changing locale that has not have translations for this key.

Reply:
It falls back to the default value
<img width="1582" alt="image" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/59447810/c655ad11-b395-430e-9b79-f208965083da">